packages/contracts/contracts/optimistic-ethereum/queue/RollupQueue.sol

```diff
@@ -54,10 +58,14 @@ contract RollupQueue {
       "Message sender does not have permission to enqueue"
     );
 
+    bytes32 txHash = keccak256(_tx);
+
     batchHeaders.push(DataTypes.TimestampedHash({
       timestamp: now,
-      txHash: keccak256(_tx)
+      txHash: txHash
     }));
+
+    emit TxEnqueued(txHash);
```

Comment:
Hmm do we need this emission? / Can we use a batch header instead?

I'm trying to reduce the event emissions to go from 3\*32 bytes to 2\*32 bytes